### ISSUE TITLE
TypeScript valid code is not generated when JavaScript identifier characters are included in the key.

### ISSUE THREAD

**Issue by kazupon** [ivanhofer/typesafe-i18n#177]
### Version

4.1.0

### Describe the bug

If you define an i18n resource whose key contains a JavaScript identifier as follows, the `i18n-types.ts` generated by `typesafe-i18n` will regenerate the code that would be an error in TypeScript.

```ts
import type { BaseTranslation } from '../i18n-types.js'

const en: BaseTranslation = {
	// TODO: your translations go here
	'HI': 'Hi {name:string}! Please leave a star if you like this project: https://github.com/ivanhofer/typesafe-i18n',
	'*': {
		'FOO': 'FOO'
	},
	'abcdefc/ghijklm': 'abcdefc/ghijklm'
}

export default en
```

In fact, TypeScript error occurs on VSCode as shown below:

<img width="1242" alt="image" src="https://user-images.githubusercontent.com/72989/154004056-cdb76162-c569-458c-ad23-cd9dfdf7b302.png">

### Reproduction

The reproduction repo is here:
https://github.com/kazupon/typesafe-i18n-repro1

```sh
git clone git@github.com:kazupon/typesafe-i18n-repro1.git
cd typesafe-i18n-repro1
pnpm i
pnpm typesafe-i18n
```

after that, you will open repo with VSCode, and open the `i18n-types.ts`

### Logs

```shell
The log for the `typesafe-i18n` command is as follows It seems to be a particularly successful log:


 % pnpm typesafe-i18n

> typesafe-i18n-repro1@1.0.0 typesafe-i18n /pato/to/repro/typesafe-i18n-repro1
> typesafe-i18n

[typesafe-i18n] version 4.1.1
[typesafe-i18n] generating files for TypeScript version: '4.5.x'
[typesafe-i18n] options: { esmImports: true }
[typesafe-i18n] watcher started in: './src/i18n/'
[typesafe-i18n] ... all files are up to date
```

### Config

```json
{
   "esmImports": true,
   "$schema": "https://unpkg.com/typesafe-i18n@4.1.1/schema/typesafe-i18n.json"
}
```
```


### Additional information

So I’m interested in it, I’ve read the typesafe-i18n code.

I think the generator should parse the key,  if JavaScript identifier chars is included in the key, the generator should be wrapped with single or double quotation.
Or, `isValidation` should be errors.
https://github.com/ivanhofer/typesafe-i18n/blob/3e566a0235cb2cc3940e5c14648b86fd249c1372/packages/generator/src/files/generate-types.ts#L218

**Comment by ivanhofer** [comment]
@kazupon thanks for reporting the issue. 
I hope you weren't to scared about the source code. The `generator` would relly need a refactoring because it's hard to understand.

The bug has be fixed here: https://github.com/ivanhofer/typesafe-i18n/blob/main/packages/generator/src/generator-util.ts#L40-L45
Currently only ` ` (space) and `-` are considered as non-valid characters in a property of an object. There is probably a JavaScript Spec where all valid characters are listed.
I need to take a deeper look so something like this will never happen in the future ;)

**Comment by ivanhofer** [comment]
Here it is: https://developer.mozilla.org/en-US/docs/Glossary/Identifier

**Comment by kazupon** [comment]
@ivanhofer 
Maybe, `esutils` package is useful for you :)
https://github.com/estools/esutils#keyword

This package has some helper APIs that is able to detect an identifier

**Comment by ivanhofer** [comment]
@kazupon thanks for the link. That's exactly what I needed :)

The issue was fixed in version `4.1.2`

**Comment by kazupon** [comment]
Thank you very much quickly! ❤️
